plugins/org.eclipse.n4js.n4jsx/src/org/eclipse/n4js/n4jsx/helpers/ReactHelper.xtend

```diff
@@ -76,21 +70,33 @@ class ReactHelper {
 	 * @param context the EObject serving the context to look for React classifiers.
 	 * @param reactClassifierName the name of React classifier.
 	 */
-	def private URI lookUpReactClassifier(EObject context, String reactClassifierName) {
+	def private TClassifier lookUpReactClassifier(EObject context, String reactClassifierName) {
 		val String key = REACT_KEY + "." + reactClassifierName;
 		return resourceScopeCacheHelper.get(key, context.eResource, [
-			// Retrieve all visible object descriptions of type TClass that are visible from 'context'
-			val descs = context.getVisibleEObjectDescriptions(TypesPackage::eINSTANCE.TClass);
-			val String reactClassifierFQNSuffix = REACT_MODULE + "." + reactClassifierName;
-			val eod = descs.findFirst[ desc | desc.isReactClassifierDescription(reactClassifierFQNSuffix) ]
-			return eod?.getEObjectURI();
+			val project = n4jsCore.findProject(context.eResource.URI).get;
+
+			var URI reactURI = null
+			// Lookup react.n4jsd in source folders
+			for (sourceContainer : project.sourceContainers) {
+				if (reactURI === null) {
+					// srcContainer is an iterable URIs
+					val ret = sourceContainer.findFirst[trimFragment.lastSegment == REACT_DEFINITION_FILE]
```

Comment:
you can directly assign to `reactURI` (no need for variable `ret`)

Reply:
done